### PR TITLE
Fixed console errors when Admin tries to switch between shops

### LINKED ISSUE #2750: [Marketplace] Get multiple console errors when switching shops
### Expected behavior
No errors

### Actual Behavior
Get a cascade of errors, probably all due to not waiting on the `Shop` subscription
```
TypeError: Cannot read property 'brandAssets' of undefined
    at Brand.getLogo (brand.js:23)
    at Brand.render (brand.js:33)
    at modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107773
    at measureLifeCyclePerf (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107053)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107772)
    at ReactCompositeComponentWrapper._renderValidatedComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107799)
    at ReactCompositeComponentWrapper._updateRenderedComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107723)
    at ReactCompositeComponentWrapper._performComponentUpdate (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107701)
    at ReactCompositeComponentWrapper.updateComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107622)
    at ReactCompositeComponentWrapper.receiveComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107524)
```

### Steps to Reproduce the Behavior
Because this is a race-condition type thing you won't always see it, but usually after a couple of switches you can.

1. Login as admin
1. Create a second shop
1. As marketplace admin switch to second shop
1. Observe errors

### Versions
```
Node: 7.10.0
NPM: 5.3.0
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 17.06.0-ce
```

### PR DESCRIPTION
**Pull request by aanchirinah**
Resolves #2750 

So on the `client` side when a shop is being fetched from the `Shops` collection by `id` it sometimes caused the issue where at times fetching a shop throws `undefined`. I resolved this by subscribing to the shops collection of types `PrimaryShop` and `MerchantShop` , before fetching for a shop from the `Shops` collection.

##### HOW TO TEST 
- Login as admin
- Create a second shop
- As marketplace admin switch to second shop
- Observe that there are no more console errors like below
```
TypeError: Cannot read property 'brandAssets' of undefined
    at Brand.getLogo (brand.js:23)
    at Brand.render (brand.js:33)
    at modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107773
    at measureLifeCyclePerf (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107053)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107772)
    at ReactCompositeComponentWrapper._renderValidatedComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107799)
    at ReactCompositeComponentWrapper._updateRenderedComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107723)
    at ReactCompositeComponentWrapper._performComponentUpdate (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107701)
    at ReactCompositeComponentWrapper.updateComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107622)
    at ReactCompositeComponentWrapper.receiveComponent (modules.js?hash=09e8e2dfa58d9bd3fd253c6052a7e955435c914c:107524)
```

### CODE REVIEW

**Comment by impactmass** [comment]
At first, the brandAssets error was gone, then I got a currency dropdown error:

```TypeError: Cannot read property 'currency' of undefined
    at findCurrency (currency.js:23)
    at _formatPriceString (currency.js:70)
    at ProductGridItems.renderGridContent (productGridItems.js:103)
    at ProductGridItems.render (productGridItems.js:144)
    at modules.js?hash=b153797acaf253e3a70e066cd5f979ffa69d762f:108778
    at measureLifeCyclePerf (modules.js?hash=b153797acaf253e3a70e066cd5f979ffa69d762f:108058)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (modules.js?hash=b153797acaf253e3a70e066cd5f979ffa69d762f:108777)
    at ReactCompositeComponentWrapper._renderValidatedComponent (modules.js?hash=b153797acaf253e3a70e066cd5f979ffa69d762f:108804)
    at ReactCompositeComponentWrapper._updateRenderedComponent (modules.js?hash=b153797acaf253e3a70e066cd5f979ffa69d762f:108728)
    at ReactCompositeComponentWrapper._performComponentUpdate (modules.js?hash=b153797acaf253e3a70e066cd5f979ffa69d762f:108706)
meteor.js?hash=6d285d84547b3dad9717a7c89c664b61b45ea3d8:942 Exception from Tracker recompute function:
meteor.js?hash=6d285d84547b3dad9717a7c89c664b61b45ea3d8:942 TypeError: Cannot read property 'language' of undefined
    at startup.js:72
    at Tracker.Computation._compute (tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:339)
    at Tracker.Computation._recompute (tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:358)
    at Object.Tracker._runFlush (tracker.js?hash=997515fa2d5b0530ba07741da556c4b36963ef3b:532)
    at onGlobalMessage (meteor.js?hash=6d285d84547b3dad9717a7c89c664b61b45ea3d8:398)```

The second time I tried, I got:
```
Warning: Failed prop type: The prop `shop` is marked as required in `Reaction(NavBar)`, but its value is `undefined`.
```
followed by the brandAssets error.

**Comment by aanchirinah** [comment]
Fixing changes requested @impactmass

**Comment by aaronjudd** [comment]
@aanchirinah @impactmass can we move this along? seems minor, but it's an annoying bug that we're seeing...

**Comment by impactmass** [comment]
Got it. We'll work together to resolve asap.

**Comment by impactmass** [comment]
Pulling this now to test again after your updates.

File: client/modules/i18n/startup.js [hunk 1]
```diff
@@ -65,10 +65,13 @@ Meteor.startup(() => {
...
       }
 
-
-      const shop = Shops.findOne({
+      let shop;
+      shop = Shops.findOne({
         _id: shopId
       });
+      if (!shop) {
+        shop = null;
```
Comment (impactmass) [inline client/modules/i18n/startup.js]: This check sets `shop` to `null`. Then immediately below it, we're referencing `shop.language`. That will throw an error. 

And that's what I got immediately I signed out: `TypeError: Cannot read property 'language' of null`

**Comment by mikemurray** [comment]
@impactmass  @aanchirinah This may be fixed in: https://github.com/reactioncommerce/reaction/pull/2966

**Pull request closed without merging**